### PR TITLE
Ignore the direction bit of the Device ID

### PR DESCRIPTION
**Pull request by megla-tlanghorst**
Fixes the extraction of the Device ID from the header. We had issues with our machine because it sets the Equipment-to-Host direction bit, which is the first bit in the 2 bytes for the Device ID. This resulted in the ID being parsed as 32769 (`0b10000000_00000001`) instead of 1 (`0b00000000_00000001`) which was configured on both sites and was correctly transmitted (with that R-Bit). This PR simply removes the first bit from the 2 Device ID bytes, since it's not part of the Device ID.

I don't think the spec is public, but this document helped me debug that issue: [Guide_to_understanding_SECS.pdf](http://www.edgeintegration.com/downloads/Guide_to_understanding_SECS.pdf)

Screenshot from that PDF:
image

### CODE REVIEW

**Comment by mkjeff** [comment]
@megla-tlanghorst  could you add a unit test for it?

**Comment by megla-tlanghorst** [comment]
I added a unit test and switched away from unsafe access as that's not necessary there (and doesn't work with arrays, despite the method taking arrays via `Span`).

I also found out that the library totally breaks down when the user has set a culture like German, that uses a different number formats, so I also fixed that while I was on it and the mask for the Steam was one bit short, so I also corrected for that.

File: test/Secs4Net.UnitTests/MessageHeader.UnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        /*
+         * Equipment to Host
+         * Device ID 1
+         * Reply expected
+         * S6F11
+         * Select request
+         * ID: 1
+         */
+        var headerBytes = new byte[] { 128, 1, 134, 11, 0, 1, 0, 0, 0, 1 };
```
Comment (mkjeff) [inline test/Secs4Net.UnitTests/MessageHeader.UnitTests.cs:18]: binary literal would be more clear, 128 => 0b10000000

**Comment by mkjeff** [comment]
I'm wondering if we need to add a new property(`EquipemntToHost`) to `MessageHeader`? If not, how does the client set the direction bit?

**Comment by megla-tlanghorst** [comment]
Already thought about that, but it might be a compatibility problem, that's why I didn't do that yet. You probably also want to set up some infrastructure to set it from the connection or so.

**Comment by mkjeff** [comment]
okay, that's keep it as. consider later until someone requests.

**Comment by megla-tlanghorst** [comment]
I think most implementations just ignore it anyways as it's peer to peer and all messages are probably directed at you anyways.

**Pull request merged**